service/matching/taskListManager.go

```diff
@@ -359,6 +359,26 @@ func (c *taskListManagerImpl) CancelPoller(pollerID string) {
 
 // Returns a batch of tasks from persistence starting form current read level.
 func (c *taskListManagerImpl) getTaskBatch() ([]*persistence.TaskInfo, error) {
+	result := make([]*persistence.TaskInfo, 0)
+	readLevel := c.taskAckManager.getReadLevel()
```

Comment:
You also want to return this readlevel back to caller so it can move the cursor when there are no tasks returned by getTaskBatch.  Otherwise we keep on doing the same work over and over again until we return a valid task for the caller to move the cursor.

Reply:
Good point, it would be more efficient in this case. Addressed